server/api/repo/create_mount.go

```diff
@@ -0,0 +1,67 @@
+package repo
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/asaskevich/govalidator"
+	"github.com/bleenco/abstruse/pkg/lib"
+	"github.com/bleenco/abstruse/server/api/middlewares"
+	"github.com/bleenco/abstruse/server/api/render"
+	"github.com/bleenco/abstruse/server/core"
+	"github.com/go-chi/chi"
+)
+
+// HandleCreateMount returns an http.HandlerFunc that writes json encoded
+// result about creating env variable to the http response body.
+func HandleCreateMount(mount core.MountsStore, repos core.RepositoryStore) http.HandlerFunc {
+	type form struct {
+		Host   string `json:"host" valid:"required"`
+		Docker string `json:"docker" valid:"required"`
+	}
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		claims := middlewares.ClaimsFromCtx(r.Context())
+		var f form
+		var err error
+		defer r.Body.Close()
+
+		id, err := strconv.Atoi(chi.URLParam(r, "id"))
+		if err != nil {
+			render.InternalServerError(w, err.Error())
+			return
+		}
+
+		if err = lib.DecodeJSON(r.Body, &f); err != nil {
+			render.InternalServerError(w, err.Error())
+			return
+		}
+
+		if valid, err := govalidator.ValidateStruct(f); err != nil || !valid {
+			render.BadRequestError(w, err.Error())
+			return
+		}
+
+		if perm := repos.GetPermissions(uint(id), claims.ID); !perm.Write {
+			render.UnathorizedError(w, "permission denied")
+			return
+		}
+
+		mnt := &core.Mount{
+			Host:         f.Host,
```

Comment:
are you checking if the host path exists? What happens if the host path does not exist?

Reply:
It fails:
```plain
[...]
==> Starting container abstruse-job-29...
Error response from daemon: invalid mount config for type "bind": bind source path does not exist: /notfound
==> rpc error: code = Unknown desc = Error response from daemon: invalid mount config for type "bind": bind source path does not exist: /notfound
```

Reply:
Should it create that directory if it doesn't exist?

Reply:
no, you should check with `fs.Exists()` and skip that mounts.